compute_endpoint/globus_compute_endpoint/endpoint/config/config.py

```diff
@@ -228,10 +227,117 @@ def __init__(
         self.log_dir = log_dir
         self.stdout = stdout
         self.stderr = stderr
-        self.debug = debug
 
-        # Misc info
-        self.display_name = display_name
+    @property
+    def executors(self) -> tuple[GlobusComputeEngineBase, ...]:
+        return self._executors
 
-        # Used to store the raw content of the YAML or Python config file
-        self.source_content: str | None = None
+    @executors.setter
+    def executors(self, val: t.Iterable[GlobusComputeEngineBase] | None):
+        if val is None:
+            self._executors: tuple[GlobusComputeEngineBase, ...] = ()
+        else:
+            self._executors = tuple(e for e in val)[:1]  # enforce len() == 0 or 1
+
+    @property
+    def heartbeat_threshold(self):
+        return self._heartbeat_threshold
+
+    @heartbeat_threshold.setter
+    def heartbeat_threshold(self, val: int):
+        self._heartbeat_threshold = max(self.heartbeat_period * 2, val)
+
+    @property
+    def idle_heartbeats_soft(self) -> int:
+        return self._idle_heartbeats_soft
+
+    @idle_heartbeats_soft.setter
+    def idle_heartbeats_soft(self, val: int):
+        self._idle_heartbeats_soft = max(0, val)
+
+    @property
+    def idle_heartbeats_hard(self) -> int:
+        return self._idle_heartbeats_hard
+
+    @idle_heartbeats_hard.setter
+    def idle_heartbeats_hard(self, val: int):
+        self._idle_heartbeats_hard = max(0, val)
+
+
+# for backwards compatibility
+Config = UserEndpointConfig
+
+
+class EndpointConfig(BaseConfig):
```

Comment:
You mean a synopsis of the configs?  Because both classes already have docstrings, describing their parameters.

Meanwhile, I'm inclined to wait on that because these classes aren't included in our documentation&nbsp;&mdash;&nbsp;but they will be when I post the PAM PR.  (Which I'm waiting to create until this PR lands.)

Reply:
> You mean a synopsis of the configs?

Yeah, it'd be nice add a brief description of the class in addition to the parameters.